api/response_test.go

```diff
@@ -97,3 +99,72 @@ func TestDecodeJSONObjects(t *testing.T) {
 	assert.Equal(t, "two", res[1].Key)
 	assert.Equal(t, "three", res[2].Key)
 }
+
+func TestAsAPIError(t *testing.T) {
+	testCases := []struct {
+		name       string
+		statusCode int
+		expected   APIError
+		shouldFail bool
+	}{
+		{
+			name:       "Not an API error (2xx)",
+			statusCode: http.StatusOK,
+			expected:   APIError{},
+			shouldFail: false,
+		},
+		{
+			name:       "Not an API error (3xx)",
+			statusCode: http.StatusNotModified,
+			expected:   APIError{},
+			shouldFail: false,
+		},
+		{
+			name:       "API error (4xx)",
+			statusCode: http.StatusNotFound,
+			expected: APIError{
+				StatusCode: http.StatusNotFound,
+				Request: rest.RequestInfo{
+					Method: http.MethodGet,
+					URL:    "https://www.dt.com/resources",
+				},
+			},
+			shouldFail: true,
+		},
+		{
+			name:       "API error (5xx)",
+			statusCode: http.StatusServiceUnavailable,
+			expected: APIError{
+				StatusCode: http.StatusServiceUnavailable,
+				Request: rest.RequestInfo{
+					Method: http.MethodGet,
+					URL:    "https://www.dt.com/resources",
+				},
+			},
+			shouldFail: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			resp := Response{
+				StatusCode: tc.statusCode,
+				Request: rest.RequestInfo{
+					Method: http.MethodGet,
+					URL:    "https://www.dt.com/resources",
+				},
+			}
+
+			err, ok := resp.AsAPIError()
+
+			assert.IsType(t, tc.expected, err)
+			assert.Equal(t, tc.expected, err)
+
+			if tc.shouldFail {
```

Comment:
this field doesn' make sense, neither in naming nor usage. I'll assume this is another reason to not trust AI double-check generated code ;) 

an `expectedOK` and simple `assert.Equal(t, tc.expectedOK, ok)` should do the trick